cutplace/fields.py

```diff
@@ -314,11 +314,29 @@ class DecimalFieldFormat(AbstractFieldFormat):
 
     def __init__(self, field_name, is_allowed_to_be_empty, length_text, rule, data_format, empty_value=None):
         super(DecimalFieldFormat, self).__init__(
-            field_name, is_allowed_to_be_empty, length_text, rule, data_format, empty_value)
-        if rule.strip() != '':
-            raise errors.InterfaceError("decimal rule must be empty")
-        self.decimalSeparator = data_format.decimal_separator
-        self.thousandsSeparator = data_format.thousands_separator
+            field_name, is_allowed_to_be_empty, "", "", data_format, empty_value)
+        # if rule.strip() != '':
+        #    raise errors.InterfaceError("decimal rule must be empty")
+        assert rule is not None, 'to specify "no rule" use "" instead of None'
+        self.decimal_separator = data_format.decimal_separator
+        self.thousands_separator = data_format.thousands_separator
+        self.valid_range = ranges.DecimalRange(rule, ranges.DEFAULT_INTEGER_RANGE_TEXT)
+        self._length = ranges.DecimalRange(length_text)
+        if self._length is not None and self._length.items is not None and len(self._length.items) == 1 \
+                and self._length.upper_limit == self._length.lower_limit:
+            self._precision = self._length.precision
+            self._scale = self._length.scale
+        else:
+            self._precision = None
+            self._scale = None
+
```

Comment:
Scratch length, scale and precision and just use the vanilla `AbstractFieldFormat.validate_length` as discussed.